src/app.js

```diff
@@ -19,22 +20,41 @@ go.app = function() {
 
     self.states.add('Facility_Code_Entry', function(name) {
       var question = $("Welcome! To report a malaria case, please enter your " +
-                       "facility code. For example, 543456");
+        "facility code. For example, 543456");
       var error = $("The facility code is invalid. Please enter again.");
       return new FreeText(name, {
         question: question,
         check: function(content) {
-          var facility_codes = self.im.config.facility_codes;
-          if (facility_codes.indexOf(parseInt(content, 10)) < 0) {
+          var facility = _.result(_.find(self.im.config.facilities, {'FacCode': content}), 'Facility');
```

Comment:
Stashing this in `im.config` results in the full 400K of JSON being read for every single interaction, regardless of whether it's needed or not. This comes with a performance hit. We need to move it into it's own config blog (as per @justinvdm's `big_config` example he posted) or into an API.